src/components/app/App.js

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import PaymentsDetails from '../payments-details/paymentsDetails';
+import {BrowserRouter, Switch} from 'react-router-dom';
+import MainContent from '../main_content/mainContent'
+
+
+import {Route} from 'react-router-dom';
+
+import './app.css';
+
+
+export default class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            users: [],
+            spendings: [],
+
+            sumOfGroupSpent : 0,
+        }
+        ;
+
+        this.onAdd = this.onAdd.bind(this);
+        this.addItem = this.addItem.bind(this);
+        this.getObjOfSpendings = this.getObjOfSpendings.bind(this);
+        this.countTotalSpendings = this.countTotalSpendings.bind(this);
+        this.onRemoveUsers = this.onRemoveUsers.bind(this);
+        this.onRemoveItems = this.onRemoveItems.bind(this);
+        this.onCancel = this.onCancel.bind(this);
+        this.getArrOfUsersSpends = this.getArrOfUsersSpends.bind(this);
+    }
+
+    async onAdd(text) {
+        let id = 1;
+        let newUser;
+        if (this.state.users.length === 0) {
```

Comment:
this condition is only due to the way to handle the `id` auto increment
do you actually need this `id`?
lets detect the proper id right on the top when you assign `1` instead of having this condition with lots of logic